src/java.base/share/classes/sun/nio/cs/StringUTF16.java

```diff
@@ -32,7 +32,7 @@ class StringUTF16 {
 
     public static char getChar(byte[] val, int index) {
         return unsafe.getChar(val,
-                              ARRAY_BYTE_BASE_OFFSET + ARRAY_BYTE_INDEX_SCALE * index * 2L);
+                (long) ARRAY_BYTE_BASE_OFFSET + ARRAY_BYTE_INDEX_SCALE * index * 2L);
```

Comment:
This expression already contains `2L` which is a `long`. So, isn't the result of the multiplications of type `long`, and consequently, isn't the entire expression calculated with `long` precision as it is?